### PR TITLE
Web Service Driver JDK11 Post Methods

### PR DESCRIPTION
**Pull request by jonreding2010**
Created Post methods
Created Post Unit Tests

### CODE REVIEW

**Review by bradyrad — COMMENTED** [review]
Some things to discuss:

- With current JMAQS Get method implementation, we just have get() and protected getContent() methods 
- MAQS Webservice implementation has the methods that have HttpResponseMessage that are named postWithResponse. JMAQS Get method version just names that get() instead. Should we mirror MAQS version naming, or continue with existing JMAQS naming?
- MAQS also has methods that just return the String version of the response, which you are including in this PR. Unsure if we want that, if we can just do a response.body() for HttpResponse(string) methods
- If we want to follow MAQS version of naming/methods, we would want to revisit Get methods

File: jmaqs-webservices-jdk11/src/main/java/com/magenic/jmaqs/webservices/jdk11/WebServiceUtilities.java [hunk 1]
```diff
@@ -112,6 +113,28 @@ public static String getResponseBody(HttpResponse<String> response) {
     return xmlMapper.writeValueAsString(body);
   }
 
+  /**
+   * Create string entity string entity.
```
[the remark this answers is not in the document]
Comment (jonreding2010) [reply jmaqs-webservices-jdk11/src/main/java/com/magenic/jmaqs/webservices/jdk11/WebServiceUtilities.java]: Updated the documentation with most recent push

File: jmaqs-webservices-jdk11/src/main/java/com/magenic/jmaqs/webservices/jdk11/WebServiceDriver.java [hunk 2]
```diff
@@ -254,7 +253,246 @@ public void setHttpRequestBuilder(HttpRequest.Builder httpRequestBuilder) {
...
+   * Execute a web service post.
+   * @param requestUri The request uri
+   * @param expectedMediaType The type of media being requested
+   * @param content The post content
+   * @param expectedStatus Assert a specific status code was returned
+   * @return The http response message
+   * @throws IOException if the exception is thrown
+   * @throws InterruptedException if the exception is thrown
+   */
+  protected HttpResponse<String> postResponse(String requestUri, MediaType expectedMediaType,
```
[the remark this answers is not in the document]
Comment (jonreding2010) [reply jmaqs-webservices-jdk11/src/main/java/com/magenic/jmaqs/webservices/jdk11/WebServiceDriver.java]: method removed

File: jmaqs-webservices-jdk11/src/main/java/com/magenic/jmaqs/webservices/jdk11/WebServiceDriver.java [hunk 3]
```diff
@@ -254,7 +253,246 @@ public void setHttpRequestBuilder(HttpRequest.Builder httpRequestBuilder) {
...
+   * Execute a web service post.
+   * @param requestUri The request uri
+   * @param expectedMediaType The type of media being requested
+   * @param content The post content
+   * @param expectSuccess Assert a success code was returned
+   * @return The http response message
+   * @throws IOException if the exception is thrown
+   * @throws InterruptedException if the exception is thrown
+   */
+  protected HttpResponse<String> postResponse(String requestUri, MediaType expectedMediaType,
```
[the remark this answers is not in the document]
Comment (jonreding2010) [reply jmaqs-webservices-jdk11/src/main/java/com/magenic/jmaqs/webservices/jdk11/WebServiceDriver.java]: method removed

File: jmaqs-webservices-jdk11/src/main/java/com/magenic/jmaqs/webservices/jdk11/WebServiceDriver.java [hunk 4]
```diff
@@ -254,7 +253,246 @@ public void setHttpRequestBuilder(HttpRequest.Builder httpRequestBuilder) {
...
+   * @param content How to encode the post content
+   * @param postMediaType The type of the media being posted
+   * @param contentAsString If true pass content as String
+   * @param expectSuccess Assert a success code was returned
+   * @return The response body as a string
+   * @throws IOException if exception is thrown
+   * @throws InterruptedException if exception is thrown
+   */
+  protected HttpResponse<String> postResponse(String requestUri, MediaType expectedMediaType, Object content,
+      MediaType postMediaType, boolean contentAsString, boolean expectSuccess)
```
[the remark this answers is not in the document]
Comment (jonreding2010) [reply jmaqs-webservices-jdk11/src/main/java/com/magenic/jmaqs/webservices/jdk11/WebServiceDriver.java]: contentAsString parameter removed

File: jmaqs-webservices-jdk11/src/main/java/com/magenic/jmaqs/webservices/jdk11/WebServiceDriver.java [hunk 5]
```diff
@@ -254,7 +253,246 @@ public void setHttpRequestBuilder(HttpRequest.Builder httpRequestBuilder) {
...
+
+    // We check for specific status
+    ensureStatusCodesMatch(response, expectedStatus);
+
+    return response;
+  }
+
+  /**
+   * Create http content.
+   * @param content The content as a string
```
[the remark this answers is not in the document]
Comment (jonreding2010) [reply jmaqs-webservices-jdk11/src/main/java/com/magenic/jmaqs/webservices/jdk11/WebServiceDriver.java:427]: created an override to the .toString()

File: jmaqs-webservices-jdk11/src/main/java/com/magenic/jmaqs/webservices/jdk11/WebServiceDriver.java [hunk 6]
```diff
@@ -254,7 +253,246 @@ public void setHttpRequestBuilder(HttpRequest.Builder httpRequestBuilder) {
...
+   * @param postMediaType The type of the media being posted
+   * @param contentAsString If true pass content as String
+   * @return The content as String
+   * @throws IOException if the exception is thrown
+   */
+  private static String createContent(Object content, MediaType postMediaType,
+      boolean contentAsString) throws IOException {
```
[the remark this answers is not in the document]
Comment (jonreding2010) [reply jmaqs-webservices-jdk11/src/main/java/com/magenic/jmaqs/webservices/jdk11/WebServiceDriver.java]: removed contentAsString Parameter and checked if Object is string type

**Review by bradyrad — APPROVED** [review]
Changes look good after updates to check for empty string response

**Pull request merged**